harvester/database/interface.py

```diff
@@ -0,0 +1,79 @@
+from sqlalchemy import create_engine, inspect
+from sqlalchemy.orm import sessionmaker, scoped_session
+from harvester.database.models import HarvestSource, HarvestJob, HarvestError
+from . import DATABASE_URI
+
+class HarvesterDBInterface:
+    def __init__(self, session=None):
+        if session is None:
+            engine = create_engine(DATABASE_URI)
+            session_factory = sessionmaker(bind=engine,
+                                           autocommit=False,
+                                           autoflush=False)
+            self.db = scoped_session(session_factory)
+        else:
+            self.db = session
+        
+    @staticmethod
+    def _to_dict(obj):
+        return {c.key: getattr(obj, c.key) 
+                for c in inspect(obj).mapper.column_attrs}
+    
+    def add_harvest_source(self, source_data):
+        new_source = HarvestSource(**source_data)
+        self.db.add(new_source)
+        self.db.commit()
+        self.db.refresh(new_source)
+        return new_source
```

Comment:
In the future, it would be nice to just accept a HarvestSource object, and do the manipulation/crosswalk here to load it.

Reply:
agree. Will do.